csharp/ql/src/Security Features/CWE-020/RuntimeChecksBypass.ql

```diff
@@ -35,9 +35,15 @@ Expr uncheckedWrite(Callable callable, Field f) {
   not callable.calls*(checkedWrite(f, _, _).getEnclosingCallable())
 }
 
+predicate serializedType(Type type) {
+  type instanceof SimpleType or
+  type instanceof StringType
+}
```

Comment:
I am not sure I understand this restriction, for example `SerializationInfo` can deserialize [arbitrary values](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.serialization.serializationinfo.getvalue?view=netframework-4.8).